cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1431,7 +1484,7 @@ func mountNonDataPartitionMatchingKernelDisk(dir, fallbacklabel string) error {
 	return doSystemdMount(partSrc, dir, opts)
 }
 
-func generateMountsCommonInstallRecover(mst *initramfsMountsState) (model *asserts.Model, sysSnaps map[snap.Type]snap.PlaceInfo, err error) {
+func generateMountsCommonInstallRecover(mst *initramfsMountsState) (model *asserts.Model, sysSnaps map[snap.Type]*seed.Snap, err error) {
```

Comment:
it probably makes sense to teach mst and the helper it uses from seed to keep the seed around so we don't have to open it again

Reply:
Did some refactoring in f1203415c9f91cdafd0ab3d70cfa20ea5d7498af.

Also change to the original commit is e600746f51f87c2b3fd214b0a152c63a6b279bf3

Reply:
I assumed that loading the essential snaps did not change the date hack, so I moved it out of loading the seed.